src/main/java/com/minecolonies/colony/jobs/JobGuard.java

```diff
@@ -0,0 +1,74 @@
+package com.minecolonies.colony.jobs;
+
+import com.minecolonies.client.render.RenderBipedCitizen;
+import com.minecolonies.colony.CitizenData;
+import com.minecolonies.entity.ai.basic.AbstractAISkeleton;
+import com.minecolonies.entity.ai.citizen.guard.EntityAIMeleeGuard;
+import com.minecolonies.entity.ai.citizen.guard.EntityAIRangeGuard;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.Random;
+
+/**
+ * Job class of the guard.
+ */
+public class JobGuard extends AbstractJob
+{
+    /**
+     * The higher the number the lower the chance to spawn a knight. Default: 3, 50% chance.
+     */
+    private static final int GUARD_CHANCE = 3;
```

Comment:
Sure let him hire/fire or kill/respawn until he got the worker he wants.
I'd like to leave all GUI stuff out of this guard version.
In the next version (another pr) I'll add a GUI to control patrolling and to switch between melee and ranger.